src/spyglass/lfp/v1/lfp_artifact_difference_detection.py

```diff
@@ -78,34 +85,48 @@ def difference_artifact_detector(
 
     # want to detect frames without parallel processing
     # compute the number of electrodes that have to be above threshold
-    nelect_above_1st = np.ceil(
-        proportion_above_thresh_1st * recording.data.shape[1]
-    )
-    nelect_above_2nd = np.ceil(
-        proportion_above_thresh_2nd * recording.data.shape[1]
-    )
+    nelect_above_1st = np.ceil(proportion_above_thresh_1st * recording.shape[1])
+    nelect_above_2nd = np.ceil(proportion_above_thresh_2nd * recording.shape[1])
+    print("num tets 1", nelect_above_1st, "num tets 2", nelect_above_2nd)
+    print("data shape", recording.shape)
 
     # find the artifact occurrences using one or both thresholds, across channels
-    # replace with LFP artifact code
-    # is this supposed to be indices??
+
     if amplitude_thresh_1st is not None:
-        # first find times with large amp change
-        artifact_boolean = np.sum(
-            (np.abs(np.diff(recording.data, axis=0)) > amplitude_thresh_1st),
-            axis=1,
+        # first find times with large amp change: sum diff over several timebins
+
+        diff_array = np.diff(recording, axis=0)
```

Comment:
The threshold is going to effectively be different in these cases because you're adding together a different number of points. A mean within the sliding window would resolve this if it matters (?).

Let me know and I can change it

Reply:
i want the threshold to be different in each case. the artifacts look different in the un-referenced data compared to the referenced.